tests/acceptance/CounterDialCept.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @group frontend
+ */
+
+$I = new AcceptanceTester( $scenario );
+
+$I->wantTo( 'create and check counter block - Progress Dial' );
+
+$slug = $I->generateRandomSlug() . 'counter';
+
+$I->havePageInDatabase(
+	[
+		'post_name'    => $slug,
+		'post_status'  => 'publish',
+		'post_content' => $I->generateGutenberg(
+			'wp:planet4-blocks/counter',
+			[
+				'title'       => 'Counter Block Progress Dial',
+				'description' => 'Testing counter block',
+				'style'       => 'arc',
+				'completed'   => '7000',
+				'target'      => '10000',
+				'text'        => '%completed% of %target% only %remaining% left',
+			]
+		),
+	]
+);
+
+// Navigate to the newly created page.
+$I->amOnPage( '/' . $slug );
+
+// Check title and description visible correctly.
+$I->see( 'Counter Block Progress Dial', '.counter-style-arc > div > header > h2' );
+$I->see( 'Testing counter block', '.counter-style-arc > div > .page-section-description p' );
+
+// Check progress dial element visible on page.
+$I->seeElement( 'div.content-counter svg.progress-arc' );
+
+// Check the text is rendered correctly.
+$I->see( '7000', 'span.counter-target' );
+$I->see( '7000 of 10000 only 3000 left', 'div.counter-text' );
```

Comment:
Maybe here also you can check the `stroke-dashoffset` attribute for the expected value of `9.45`.